bitcoin/src/psbt/mod.rs

```diff
@@ -38,15 +38,54 @@ pub use self::map::{Input, Output, PsbtSighashType};
 /// Partially signed transaction, commonly referred to as a PSBT.
 pub type Psbt = PartiallySignedTransaction;
 
+/// Future version of PSBT this library can't parse
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+pub struct FutureVersionError(u32);
+
+impl fmt::Display for FutureVersionError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "PSBT version {} is not supported", self.0)
+    }
+}
+
+#[cfg(feature = "std")]
+impl ::std::error::Error for FutureVersionError {}
+
+/// Partially signed transaction version as defined in BIP-174 and BIP-370.
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde", serde(crate = "actual_serde"))]
+#[non_exhaustive]
+pub enum Version {
+    /// PSBT version 0, defined in BIP-174
+    PsbtV0 = 0,
+    /// PSBT version 2, defined in BIP-370
+    PsbtV2 = 2,
+}
+
+impl Version {
+    fn to_raw(self) -> u32 {
+        self as u32
+    }
+
+    fn from_raw(version: u32) -> Result<Version, FutureVersionError> {
+        match version {
+            v if v == Version::PsbtV0 as u32 => Ok(Version::PsbtV0),
+            v if v == Version::PsbtV2 as u32 => Ok(Version::PsbtV2),
+            future => Err(FutureVersionError(future))
+        }
```

Comment:
I don't have deep PSBT knowledge but this reads as if someone attempting to parse 0 for `version` will get a future version error but v1 will never exist, right?

Reply:
0 will get PsbtV0
1 will get Future version which I agree is weird. Maybe just VersionError is more appropriate

Reply:
Never say never ;).

But I agree, I think we should have a specific error variant for v1 actually which tells the user that they probably meant v0.